src/components/popover-container/popover-container.stories.js

```diff
@@ -12,18 +20,280 @@ export default {
   }
 };
 
-export const basic = () => {
+const storyStyle = (height = '80px', position = 'right') => ({
+  minHeight: height,
+  marginLeft: position === 'left' ? '400px' : null
+});
+
+export const Basic = () => {
   const title = text('title', 'Popover Title');
   const iconType = select('iconType', [...OptionsHelper.icons], 'settings');
   const position = select('position', [...OptionsHelper.alignBinary], 'right');
+  const isStickyTop = boolean('isStickyTop', true);
+
+  const [isOpen, setOpen] = useState(false);
+  const iconRef = useRef(null);
+
+  const handleClick = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    iconRef.current.focus();
+  };
 
   return (
-    <div style={ position === 'left' ? { marginLeft: '400px' } : null }>
+    <div style={ storyStyle(undefined, position) }>
       <PopoverContainer
         title={ title }
         position={ position }
-        iconType={ iconType }
+        isOpen={ isOpen }
+        onClose={ handleClose }
+        isStickyTop={ isStickyTop }
+        renderOpenComponent={ () => (
+          <IconButton
+            ref={ iconRef }
+            data-element='popover-container-icon'
+            onAction={ handleClick }
+          >
+            <Icon type={ iconType } />
+          </IconButton>
+        ) }
+      />
+    </div>
+  );
+};
+
+export const Filter = () => {
+  const initValues = [
+    {
+      value: 'Option 1',
+      checked: false
+    },
+    {
+      value: 'Option 2',
+      checked: false
+    },
+    {
+      value: 'Option 3',
+      checked: false
+    }
+  ];
+
+  const [isOpen, setOpen] = useState(false);
+  const [options, setOptions] = useState(initValues);
+  const [filters, setFilters] = useState([]);
+
+  const clearAllOptions = () => {
+    const temps = options;
+
+    for (let i = 0; i < temps.length; i++) {
+      temps[i].checked = false;
+    }
+
+    setOptions([...temps]);
+  };
+
+  const clearFilters = () => {
+    setFilters([]);
+  };
+
+  const updateCheckValue = (e) => {
+    const temps = options;
+
+    for (let i = 0; i < temps.length; i++) {
```

Comment:
Could potentially rewrite like this but fine if you want to keep it as you have it already:
```suggestion
    const index = temps.findIndex(temp => temp.value === e.target.value);
    if (index !== -1) {
        temps[i].checked = !temps[i].checked;
        setOptions([...temps]);
    }
```

Reply:
Yes, I think that your change is more modern. I'm going to use this